src/main/java/io/github/sashirestela/openai/OpenAI.java

```diff
@@ -69,6 +73,25 @@
  */
 public interface OpenAI {
 
+    static ChatRequest updateRequest(ChatRequest chatRequest, Boolean useStream) {
+        var toolChoice = chatRequest.getToolChoice();
+
+        if (!isNullOrEmpty(chatRequest.getTools())) {
+            if (toolChoice == null) {
+                toolChoice = ChatToolChoiceType.AUTO;
+            } else if (!(toolChoice instanceof ChatToolChoice) &&
```

Comment:
@the-gigi Why have you brought this type validation code? This should keep on the ChatRequest class